toolkit/offchain/src/reserve/release.rs

```diff
@@ -96,20 +95,21 @@ fn reserve_release_tx(
 	reserve_data: &ReserveData,
 	previous_reserve: &ReserveUtxo,
 	reference_utxo: &OgmiosUtxo,
-	cumulative_total_transfer: u64,
+	amount_to_transfer: u64,
 	latest_slot: u64,
 	costs: Costs,
 ) -> anyhow::Result<Transaction> {
-	let token = previous_reserve.datum.immutable_settings.token.clone();
-	let stats = previous_reserve.datum.stats.clone();
+	let token = &previous_reserve.datum.immutable_settings.token;
+	let stats = &previous_reserve.datum.stats;
 
 	let mut tx_builder = TransactionBuilder::new(&get_builder_config(ctx)?);
 
 	let reserve_balance = previous_reserve.utxo.get_asset_amount(&token);
 	let token_total_amount_transferred = stats.token_total_amount_transferred;
-	let amount_to_transfer = cumulative_total_transfer.checked_sub(token_total_amount_transferred)
-		.filter(|x| *x > 0)
-		.ok_or_else(||anyhow!("The requested total amount: {cumulative_total_transfer} is not greater than the amount already transferred to illiquid supply: {token_total_amount_transferred}.",))?;
+	let cumulative_total_transfer: u64 = token_total_amount_transferred
+		.checked_add(amount_to_transfer)
+		.expect("cumulative_total_transfer overflow u64");
```

Comment:
I know this can be controversial. Problem is that with the current CSL it will be really hard to make it u128. In other words our implementation is not ready for going over u64. Therefore panic.